pallets/author-mapping/src/migrations.rs

```diff
@@ -14,17 +14,86 @@
 // You should have received a copy of the GNU General Public License
 // along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
 
-use crate::{BalanceOf, Config, MappingWithDeposit, RegistrationInfo};
+use crate::{BalanceOf, Config, Event, MappingWithDeposit, NimbusLookup, Pallet, RegistrationInfo};
 use frame_support::{
 	pallet_prelude::PhantomData,
-	traits::{Get, OnRuntimeUpgrade},
+	traits::{Get, OnRuntimeUpgrade, ReservableCurrency},
 	weights::Weight,
 };
 use nimbus_primitives::NimbusId;
 use parity_scale_codec::{Decode, Encode};
 #[cfg(feature = "try-runtime")]
 use scale_info::prelude::format;
 
+/// Migrates MappingWithDeposit map value from RegistrationInfo to RegistrationInformation,
+/// thereby adding a keys: T::Keys field to the value to support VRF keys that can be looked up
+/// via NimbusId.
+pub struct AddAccountIdToNimbusLookup<T>(PhantomData<T>);
+impl<T: Config> OnRuntimeUpgrade for AddAccountIdToNimbusLookup<T> {
+	fn on_runtime_upgrade() -> Weight {
+		log::info!(target: "AddAccountIdToNimbusLookup", "running migration");
+
+		let mut read_count = 0u64;
+		let mut write_count = 0u64;
+		<MappingWithDeposit<T>>::translate(|nimbus_id, registration_info: RegistrationInfo<T>| {
+			read_count += 2u64;
+			if NimbusLookup::<T>::get(&registration_info.account).is_none() {
+				<NimbusLookup<T>>::insert(&registration_info.account, nimbus_id);
+				write_count += 2u64;
+				Some(registration_info)
+			} else {
+				// revoke the additional association and return the funds
+				T::DepositCurrency::unreserve(
```

Comment:
Doesnt unreserve involve writes too? Might be that these are not counted?

Reply:
Good point, counted `unreserve` as 1 write